awsx/ec2/subnetDistributorNew.ts

```diff
@@ -0,0 +1,296 @@
+// Copyright 2016-2022, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Code adapted from https://github.com/jen20/pulumi-aws-vpc/blob/master/nodejs/src/subnetDistributor.ts
+// and used in accordance with MPL v2.0 license
+
+import * as pulumi from "@pulumi/pulumi";
+import { SubnetSpecInputs, SubnetTypeInputs } from "../schema-types";
+import { Netmask } from "netmask";
+
+export interface SubnetSpec {
+  cidrBlock: string;
+  type: SubnetTypeInputs;
+  azName: string;
+  subnetName: string;
+  tags?: pulumi.Input<{
+    [key: string]: pulumi.Input<string>;
+  }>;
+}
+
+export function getSubnetSpecs(
+  vpcName: string,
+  vpcCidr: string,
+  azNames: string[],
+  subnetInputs: SubnetSpecInputs[] | undefined,
+): SubnetSpec[] {
+  const vpcNetmask = new Netmask(vpcCidr);
+  const newBitsPerAZ = Math.log2(nextPow2(azNames.length));
```

Comment:
Isn't `ceil(log2(length))` slightly clearer here? Actually if you use that it looks like you can get rid of the `nextPow2` function.

Reply:
Hmm, these are nearly the same, except for the `0` case. I think this might be better just being a dedicated function that wraps `ceil(log2(x))` instead though.